module/documents/item.mjs

```diff
@@ -605,16 +605,22 @@ export default class Item5e extends Item {
    * Retrieve an item's critical hit threshold. Uses the smallest value from among the following sources:
    * - item document
    * - item document's actor (if it has one)
+   * - item document's ammunition (if it has any)
    * - the constant '20'
    * @returns {number|null}  The minimum value that must be rolled to be considered a critical hit.
    */
   getCriticalThreshold() {
     const actorFlags = this.actor.flags.dnd5e || {};
     if ( !this.hasAttack ) return null;
     let actorThreshold = null;
+    let it = this.system.critical?.threshold ?? 20;
+    let at = 20;
```

Comment:
Perhaps it would be better to use more descriptive variables names, `itemThreshold` and `ammoThreshold`?